File: package/ios/expo/views/SmileIDExpoSmartSelfieCaptureView.swift
import ExpoModulesCore
import UIKit
import SwiftUI
import SmileID

/**
 * Expo implementation of SmartSelfieCapture using the new architecture
 */
class SmileIDExpoSmartSelfieCaptureView: SmileIDExpoView {
    
    var showConfirmation: Bool = true
    var useStrictMode: Bool = false
    
    override func renderContent() {
        let config = createConfig()
        
        // Use SmileIDExtReactNative to create the view
        let uiView = SmileID.RNSmartSelfieCapture(
            config: config,
            onResult: { [weak self] result in
                self?.handleResultCallback(result)
            }
        )
        
        // Use the base class helper to set the UIView
        setUIView(uiView)
    }
    
    private func createConfig() -> SmileIDViewConfig {
        var config = SmileIDViewConfig()
        config.userId = userId
        config.jobId = jobId
        config.allowAgentMode = allowAgentMode
        config.showInstructions = showInstructions
        config.showAttribution = showAttribution
        config.extraPartnerParams = extraPartnerParams
        config.showConfirmation = showConfirmation
        config.useStrictMode = useStrictMode
        // Always skip API submission for capture
        config.skipApiSubmission = true
        return config
    }
}

File: package/android/src/expo/com/smileidentity/expo/views/SmileIDExpoSmartSelfieCaptureView.kt
package com.smileidentity.expo.views

import android.content.Context
import androidx.compose.runtime.Composable
import com.smileidentity.SmileID
import com.smileidentity.shared.RNSmartSelfieCapture
import com.smileidentity.shared.SmileIDViewConfig
import expo.modules.kotlin.AppContext

/**
 * Example Expo implementation of SmartSelfieCapture using the new architecture
 */
class SmileIDExpoSmartSelfieCaptureView(
    context: Context,
    appContext: AppContext
) : SmileIDExpoView(context, appContext) {

    var showConfirmation: Boolean = true
    var useStrictMode: Boolean = false

    override fun renderContent() {
        setContentWithTheme {
            RenderSmartSelfieCaptureContent()
        }
    }

    @Composable
    private fun RenderSmartSelfieCaptureContent() {
      val config = SmileIDViewConfig(
        userId = userId,
        jobId = jobId,
        allowAgentMode = allowAgentMode ?: false,
        showInstructions = showInstructions,
        skipApiSubmission = skipApiSubmission,
        showAttribution = showAttribution,
        extraPartnerParams = extraPartnerParams,
        showConfirmation = showConfirmation,
        useStrictMode = useStrictMode
      )

      SmileID.RNSmartSelfieCapture(
        config = config,
        onResult = { result ->
          handleResultCallback(result)
        }
      )
    }
}
